pgx/src/datum/time.rs

```diff
@@ -121,34 +112,13 @@ impl serde::Serialize for Time {
     where
         S: serde::Serializer,
     {
-        let (h, m, s, micro) = self.clone().to_hms_micro();
-        let t = time::Time::from_hms_micro(h, m, s, micro).unwrap();
-        if t.millisecond() > 0 {
-            serializer.serialize_str(
-                &t.format(
-                    &time::format_description::parse(&format!(
-                        "[hour]:[minute]:[second].{}",
-                        t.millisecond()
-                    ))
-                    .map_err(|e| {
-                        serde::ser::Error::custom(format!("Time invalid format problem: {:?}", e))
-                    })?,
-                )
-                .map_err(|e| {
-                    serde::ser::Error::custom(format!("Time formatting problem: {:?}", e))
-                })?,
-            )
-        } else {
-            serializer.serialize_str(&t.format(&DEFAULT_TIME_FORMAT).map_err(|e| {
-                serde::ser::Error::custom(format!("Time formatting problem: {:?}", e))
-            })?)
-        }
+        let cstr: Option<&cstr_core::CStr> = unsafe {
+            crate::direct_function_call(pg_sys::time_out, vec![self.clone().into_datum()])
```

Comment:
I think ideally we'd return a serde error if we're on the wrong thread (rather than panic), since it may be surprising that this calls into postgres in a `Serialize` impl.

But we don't provide a function to test this, and doubtlessly we have tons of places with the same kinds of caveats.

Reply:
Yeah, I am thinking: is `check_active_thread` okay for users to call? Or should we have `pgx_pg_sys` expose something like `direct_funcall_or_fail` that returns a `Result` with the Err variant indicating the multithreading scenario?